.github/workflows/pull-request-ci.yaml

```
@@ -0,0 +1,307 @@
+    name: tests
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    needs: build
+    env:
+      # For KubeCF
+      PINNED_TOOLS: true
+      TOOLS_DIR: ${{ github.workspace }}/tools
+      FEATURE_AUTOSCALER: "true"
+      # For catapult
+      # Note that credentials-related parts are in the individual steps
+      BACKEND: gke
+      DOWNLOAD_CATAPULT_DEPS: "false"
+
+    defaults:
+      run:
+        working-directory: kubecf
+
+    strategy:
+      matrix:
+        include:
+        - backend: diego
+          experimental: false
+        - backend: eirini
+          experimental: true
+
+    steps:
+    - name: Checkout KubeCF
+      uses: actions/checkout@v2
+      with:
+        path: kubecf
+        submodules: recursive
+    - name: Checkout catapult
+      uses: actions/checkout@v2
+      with:
+        repository: SUSE/catapult
+        path: catapult
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: cache.tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/tools
+        key: ${{ runner.os }}-tools
+    - run: make tools-install
+
+    - id: download-bundle
+      name: Download chart bundle
+      uses: actions/download-artifact@v2
+      with:
+        name: kubecf-bundle.tgz
+        # The artifact upload/download creates an extra directory
+        path: ${{ github.workspace }}/kubecf-bundle
+
+    - name: Download gomplate
+      run: |
+        if test -x "${TOOLS_DIR}/gomplate" ; then
+          echo "gomplate already exists; skipping."
+          exit 0
+        fi
+        mkdir -p "${TOOLS_DIR}"
+        wget -O "${TOOLS_DIR}/gomplate" \
+          https://github.com/hairyhenderson/gomplate/releases/latest/download/gomplate_linux-amd64
+        chmod a+x "${TOOLS_DIR}/gomplate"
+
+    - name: Start SSH agent
+      run: |
+        set -o errexit -o pipefail -o nounset
+        eval "$(ssh-agent -s)"
+        ssh-keygen -t rsa -b 4096 -N '' -C "KubeCF CI #${{ github.run_id }}" -f ssh-key
+        ssh-add ssh-key
+        echo "::add-mask::$(cat ssh-key.pub)"
+        rm -f ssh-key ssh-key.pub
+        echo "::set-env name=SSH_AUTH_SOCK::${SSH_AUTH_SOCK}"
+        echo "::set-env name=SSH_AGENT_PID::${SSH_AGENT_PID}"
+
+    - name: set GKE_CRED_JSON
+      run: |
+        json_file="$(mktemp)"
+        echo "$GKE_CRED_JSON" > "${json_file}"
+        echo "::set-env name=GKE_CRED_JSON::${json_file}"
+      env:
+        GKE_CRED_JSON: ${{ secrets.GKE_CREDS_JSON }}
+
+    - name: cache.catapult-common-tools
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.workspace }}/catapult/modules/common/bin
+        key: ${{ runner.os }}-catapult-common-tools
+
+    - run: make k8s
+      working-directory: catapult
+      env:
+        GKE_CLUSTER_NAME: kubecf-ci-${{ github.run_id }}-${{ matrix.backend }}
+        OWNER: ${{ github.repository_owner }}
+        DEBUG_MODE: false
+        GKE_NODE_COUNT: 1
```

```suggestion
        GKE_NODE_COUNT: 1
        GKE_PREEMPTIBLE: true
```

We should also set the cluster as preemptible, to mirror the concourse setup.